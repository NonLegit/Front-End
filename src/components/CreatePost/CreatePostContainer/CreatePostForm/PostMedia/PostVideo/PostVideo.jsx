import DeleteIcon from '@mui/icons-material/Delete';
import { Box } from '@mui/material';
import { CustomIconButton } from './styles';
/**
 * This component is post video
 *
 * @component PostVideo
 * @property {string} src -video source.
 * @property {function} setPostMedia -To set post media
 * @returns {React.Component} PostVideo
 */
/* eslint-disable jsx-a11y/media-has-caption */
function PostVideo(props) {
  const { src, setPostMedia } = props;
  const handleClick = () => {
    setPostMedia([]);
  };
  return (
    <Box
      display="flex"
      flexDirection="column"
      border={1}
      borderRadius={1}
      borderColor="#edeff1"
    >
      <video
        controls
        style={{ width: '100%', maxHeight: '512px' }}
      >
        <source src={src} type="video/mp4" />
      </video>
      <CustomIconButton
        color="third"
        onClick={handleClick}
      >
        <DeleteIcon />
      </CustomIconButton>
    </Box>
  );
}

export default PostVideo;
