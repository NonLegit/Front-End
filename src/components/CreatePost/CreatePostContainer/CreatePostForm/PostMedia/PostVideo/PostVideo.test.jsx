import renderer from 'react-test-renderer';
import PostVideo from './PostVideo';

describe(PostVideo, () => {
  it('PostVideo renders correctly', () => {
    const tree = renderer
      .create(<PostVideo src="" setPostMedia={(arr) => arr} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
