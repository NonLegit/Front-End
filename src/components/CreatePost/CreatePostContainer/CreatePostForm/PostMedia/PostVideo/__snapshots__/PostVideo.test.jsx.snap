// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostVideo PostVideo renders correctly 1`] = `
<div
  className="MuiBox-root css-m1xdaq"
>
  <video
    controls={true}
    style={
      Object {
        "maxHeight": "512px",
        "width": "100%",
      }
    }
  >
    <source
      src=""
      type="video/mp4"
    />
  </video>
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorThird MuiIconButton-sizeMedium css-jct5ou-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="DeleteIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
      />
    </svg>
  </button>
</div>
`;
